src/propagators/zipkin_b3_text_map_codec.js

```diff
@@ -0,0 +1,157 @@
+        }
+
+        return value;
+    }
+
+    _decodeURIValue(value: string): string {
+        // unfortunately, decodeURIComponent() can throw 'URIError: URI malformed' on bad strings
+        try {
+            return decodeURIComponent(value);
+        } catch (e) {
+            return value;
+        }
+    }
+
+    extract(carrier: any): ?SpanContext {
+        let spanContext = new SpanContext();
+        let baggage = {};
+
+        for (let key in carrier) {
+            if (carrier.hasOwnProperty(key)) {
+                let lowerKey = key.toLowerCase();
+
+                switch (lowerKey) {
+                    case ZIPKIN_PARENTSPAN_HEADER:
+                        spanContext.parentId = this._decodeValue(carrier[ZIPKIN_PARENTSPAN_HEADER]);
+                        break;
+                    case ZIPKIN_SPAN_HEADER:
+                        spanContext.spanId = this._decodeValue(carrier[ZIPKIN_SPAN_HEADER]);
+                        break;
+                    case ZIPKIN_TRACE_HEADER:
+                        spanContext.traceId = this._decodeValue(carrier[ZIPKIN_TRACE_HEADER]);
+                        break;
+                    case ZIPKIN_SAMPLED_HEADER:
+                        spanContext.flags = spanContext.flags | constants.SAMPLED_MASK;
+                        break;
+                    case ZIPKIN_DEBUG_HEADER:
+                        // "debug implies sampled"
+                        // https://github.com/openzipkin/b3-propagation
+                        spanContext.flags = spanContext.flags | constants.SAMPLED_MASK | constants.DEBUG_MASK;
+                        break;
+                    case constants.JAEGER_DEBUG_HEADER:
+                        spanContext.debugId = this._decodeValue(carrier[constants.JAEGER_DEBUG_HEADER]);
+                        break;
+                    case constants.JAEGER_BAGGAGE_HEADER:
+                        this._parseCommaSeparatedBaggage(baggage, this._decodeValue(carrier[key]));
+                        break;
+                    default:
+                        if (Utils.startsWith(lowerKey, this._baggagePrefix)) {
+                            let keyWithoutPrefix = key.substring(this._baggagePrefix.length);
+                            baggage[keyWithoutPrefix] = this._decodeValue(carrier[key]);
+                        }
+                }
+            }
+        }
+
+        spanContext.baggage = baggage;
+        return spanContext;
+    }
+
+    inject(spanContext: SpanContext, carrier: any): void {
+        carrier[ZIPKIN_TRACE_HEADER] = spanContext.traceIdStr;
+        carrier[ZIPKIN_PARENTSPAN_HEADER] = spanContext.parentIdStr;
+        carrier[ZIPKIN_SPAN_HEADER] = spanContext.spanIdStr;
+
+        // > Since Debug implies Sampled, so don't also send "X-B3-Sampled: 1"
+        // https://github.com/openzipkin/b3-propagation
+
+        if (spanContext.isDebug()) {
+           carrier[ZIPKIN_DEBUG_HEADER] = '1';
+        } else {
```

Comment:
why `else` rather than independently?

Reply:
Based on [the zipkin propagation spec](https://github.com/openzipkin/b3-propagation#debug-flag):

> Debug is encoded as X-B3-Flags: 1. Since Debug implies Sampled, so don't also send "X-B3-Sampled: 1".

So, the `inject` call only sets the sampled header if debug is false.  I've modified the code to make this a little more clear.